### PR TITLE
Remove codegangsta/cli and its replace directive with urfave/cli

### PR DESCRIPTION
**Pull request by andy-a-d-nguyen**
- [x] Read the [Contributing document](../blob/-/.github/CONTRIBUTING.md).

Summary
---------------
A replace directive is being used [here](https://github.com/cloudfoundry/routing-release/blob/bc125319b8be3c3b405ed932b40929ef500b9501/src/code.cloudfoundry.org/go.mod#L9) to pin `codegangsta/cli`'s version. [codegangsta/cli](https://github.com/cloudfoundry-incubator/spiff/blob/master/vendor/github.com/codegangsta/cli/README.md) has been archived, but [urfave/cli](https://github.com/urfave/cli) has the same APIs while still being maintained and `routing-release` also currently uses `negroni`, another package from `urfave` [here](https://github.com/cloudfoundry/routing-release/blob/bc125319b8be3c3b405ed932b40929ef500b9501/src/code.cloudfoundry.org/go.mod#L48). https://github.com/codegangsta/cli also redirects to https://github.com/urfave/cli

This PR removes the replace directive for `codegangsta/cli` as well as replace the package with `urfave/cli`

Backward Compatibility
---------------
Breaking Change? **No**

### CODE REVIEW

**Comment by andy-a-d-nguyen** [comment]
Tracker issue #187603705

**Comment by maxmoehl** [comment]
Minor correction: I think the repo was transferred causing the module to also change names. At least GitHub has a redirect for this repo:
```
$ curl -i https://github.com/codegangsta/cli
HTTP/2 301
server: GitHub.com
date: Mon, 08 Jul 2024 09:01:51 GMT
location: https://github.com/urfave/cli
[...]
```

**Review by maxmoehl — APPROVED** [review]
LGTM, not sure if we should merge right now, the CI is red since a few days :/

**Pull request merged**
